pages/establishment/update/index.js

```diff
@@ -1,11 +1,36 @@
-const { omit } = require('lodash');
+const { pick, uniq } = require('lodash');
 const { page } = require('@asl/service/ui');
 const { form } = require('../../common/routers');
 const schema = require('./schema');
 const confirm = require('./routers/confirm');
 const success = require('../../common/routers/success');
 const { groupFlags, ungroupFlags } = require('../../establishment/formatters/flag-grouping');
 
+const fieldsToAuthorisations = params => {
+  return Object.keys(params).reduce((authorisations, fieldName) => {
+    if (params[fieldName] && /^authorisation-/.test(fieldName)) {
+      const id = fieldName.slice(-36); // get the uuid
```

Comment:
🤢

Reply:
it sucks but I can't think of another way - any suggestions?

Reply:
I would at the very least validate that the substring actually _is_ a UUID, to avoid the possibility of me POSTing fields named `authorisation-drop table projects;` or similar. At the moment you're just trusting that the last 36 characters of any field with the right prefix are valid.

Reply:
Or match against `/^[0-9A-F]{8}-[0-9A-F]{4}-4[0-9A-F]{3}-[89AB][0-9A-F]{3}-[0-9A-F]{12}$/` in the regex.

Reply:
```js
const re = /^authorisation-([0-9a-f]{8}-[0-9a-f]{4}-4[0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12})$/;
const matched = fieldName.match(re);
if (matched) {
  const id = matched[1]; // I think, double check this
  /// etc
}
```

Reply:
There's other shizzle in the middle so i'll just confirm it's a uuid before using the field.

Reply:
Lenny's solution is a _lot_ better than the .slice(-36) - could we use a variation of this?

Reply:
There's now a check below for uuid but ok